include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -581,7 +593,7 @@ struct __get_sycl_range
         assert(__first < __last);
         using value_type = val_t<_Iter>;
 
-        const auto __offset = __first - oneapi::dpl::begin(__first.get_buffer());
+        const auto __offset = __first.get_idx();
```

Comment:
I think it is equivalent to code that we had before.

`const auto __offset = __first - oneapi::dpl::begin(__first.get_buffer());`

Reply:
Similar response to above, it is the same as long as `__first` is a `sycl_iterator`.  If it is instead a `dpct::device_pointer` then this subtraction is undefined.

Reply:
I see.
It looks we need to extend the contract for sycl_iterator type